tests/test_surrogates.py

```diff
@@ -0,0 +1,303 @@
+            self.assertAlmostEqual(surrogate_results.states[i]['falling'], result.event_states[i]['falling'], delta=0.1)
+            self.assertEqual(surrogate_results.states[i]['falling'], surrogate_results.event_states[i]['falling'])
+            self.assertAlmostEqual(surrogate_results.states[i]['impact'], result.event_states[i]['impact'], delta=0.1)
+            self.assertEqual(surrogate_results.states[i]['impact'], surrogate_results.event_states[i]['impact'])
+        
+        # State subset
+        surrogate = m.generate_surrogate([load_eqn], dt = 0.1, save_freq = 0.25, threshold_keys = 'impact', states = ['v'])
+        self.assertListEqual(surrogate.states, ['v'] + m.outputs + m.events + m.inputs)
+        self.assertListEqual(surrogate.inputs, m.inputs)
+        self.assertListEqual(surrogate.outputs, m.outputs)
+        self.assertListEqual(surrogate.events, m.events)
+
+        surrogate = m.generate_surrogate([load_eqn], dt = 0.1, save_freq = 0.25, threshold_keys = 'impact', states = 'v')
+        self.assertListEqual(surrogate.states, ['v'] + m.outputs + m.events + m.inputs)
+        self.assertListEqual(surrogate.inputs, m.inputs)
+        self.assertListEqual(surrogate.outputs, m.outputs)
+        self.assertListEqual(surrogate.events, m.events)
+
+        options = {
+            'threshold_keys': 'impact',
+            'save_freq': 0.25,
+            'dt': 0.25
+        }
+        result = m.simulate_to_threshold(load_eqn, **options)
+        surrogate_results = surrogate.simulate_to_threshold(load_eqn, **options)
+        self.assertAlmostEqual(surrogate_results.times[-1], result.times[-1], delta=0.26)
+        for i in range(min(len(result.times), len(surrogate_results.times))):
+            self.assertListEqual(list(surrogate_results.inputs[i].keys()), list(surrogate_results.inputs[i].keys()))
+            self.assertAlmostEqual(surrogate_results.states[i]['x'], result.states[i]['x'], delta=8)
+            self.assertEqual(surrogate_results.states[i]['x'], surrogate_results.outputs[i]['x'])
+            self.assertAlmostEqual(surrogate_results.states[i]['v'], result.states[i]['v'], delta=1)
+            self.assertAlmostEqual(surrogate_results.states[i]['falling'], result.event_states[i]['falling'], delta=0.1)
+            self.assertEqual(surrogate_results.states[i]['falling'], surrogate_results.event_states[i]['falling'])
+            self.assertAlmostEqual(surrogate_results.states[i]['impact'], result.event_states[i]['impact'], delta=0.1)
+            self.assertEqual(surrogate_results.states[i]['impact'], surrogate_results.event_states[i]['impact'])
+
+        # Events subset
+        surrogate = m.generate_surrogate([load_eqn], dt = 0.1, save_freq = 0.25,      threshold_keys = 'impact', events = ['impact'])
+        surrogate = m.generate_surrogate([load_eqn], dt = 0.1, save_freq = 0.25,      threshold_keys = 'impact', events = 'impact')
```

Comment:
Should both of these two lines (209 and 210) be "surrogate = "?

Reply:
One is doing it with events as a list, the other one as a string. We don't test anything except to see that no exception was raised. 

I guess there is an untested assumption that the first surrogate is identical to the second. Let me think about how to test that